caps/typeattr_path_test.go

```diff
@@ -0,0 +1,132 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package caps
+
+import (
+	"fmt"
+	"regexp"
+
+	. "gopkg.in/check.v1"
+)
+
+type pathSuite struct {
+	pathAttr    TypeAttr
+	devPathAttr TypeAttr
+	capType     *Type
+	cap         *Capability
+}
+
+var _ = Suite(&pathSuite{})
+
+func (s *pathSuite) SetUpSuite(c *C) {
+	// A path attribute that accepts any path
+	s.pathAttr = &pathAttr{
+		errorHint:       "path cannot be empty",
+		allowedPatterns: []*regexp.Regexp{regexp.MustCompile(".*")},
+	}
+	// A path attribute restricted to files in /dev/
+	s.devPathAttr = &pathAttr{
+		errorHint:       "devPath must point to something in /dev/",
+		allowedPatterns: []*regexp.Regexp{regexp.MustCompile("^/dev/.+$")},
+	}
+	// An test capability and capability type using attributes
+	s.capType = &Type{
+		Name: "type",
+		Attrs: map[string]TypeAttr{
+			"path":    s.pathAttr,
+			"devPath": s.devPathAttr,
+		},
+	}
+}
+
+func (s *pathSuite) SetUpTest(c *C) {
+	// For testing, mock the filepath.EvalSymlinks function to a simple
+	// identity function that never fails. Individual tests can replace that
+	// with anything they want, the tear-down function recovers the original.
+	MockEvalSymlinks(IgnoreSymbolicLinks)
```

Comment:
There is a testutils/base.go:BaseTEst.AddCleanup(). So you could write MockEvalSymlinks in a way that it adds the cleanup handler itself and you avoid the need to remind users that they need to reset it in TearDownTests. But not important, just noticed while reading this bit.